[Bug] Failed to build BE on macOS

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

Doris Version: master
OS: macOS 13.5.2
Compiler:
```shell
Homebrew clang version 16.0.6
Target: arm64-apple-darwin22.6.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
```


### What's Wrong?

```shell
[955/955] Linking CXX executable src/service/doris_be
FAILED: src/service/doris_be
: && /opt/homebrew/opt/llvm@16/bin/clang++ -O0 -fsanitize=address -DADDRESS_SANITIZER  -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk -mmacosx-version-min=13.5 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  src/service/CMakeFiles/doris_be.dir/doris_main.cpp.o -o src/service/doris_be -F/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/System/Library/Frameworks src/agent/libAgent.a  src/common/libCommon.a  src/exec/libExec.a  src/exprs/libExprs.a  src/gutil/libGutil.a  src/io/libIO.a  src/olap/libOlap.a  src/runtime/libRuntime.a  src/service/libService.a  src/udf/libUdf.a  src/util/libUtil.a  src/gen_cpp/libDorisGen.a  src/http/libWebserver.a  src/geo/libGeo.a  src/geo/libGeoType.a  src/vec/libVec.a  src/pipeline/libPipeline.a  /Programs/doris/thirdparty/installed/lib64/libboost_date_time.a  /Programs/doris/thirdparty/installed/lib/libgflags.a  /Programs/doris/thirdparty/installed/lib/libglog.a  /Programs/doris/thirdparty/installed/lib/libbacktrace.a  /Programs/doris/thirdparty/installed/lib/libre2.a  /Programs/doris/thirdparty/installed/lib64/libhs.a  /Programs/doris/thirdparty/installed/lib/libodbc.a  /Programs/doris/thirdparty/installed/gperftools/lib/libprofiler.a  /Programs/doris/thirdparty/installed/lib/libprotobuf.a  /Programs/doris/thirdparty/installed/lib/libgtest.a  /Programs/doris/thirdparty/installed/lib/libgtest_main.a  /Programs/doris/thirdparty/installed/lib/libbenchmark.a  /Programs/doris/thirdparty/installed/lib/libgmock.a  /Programs/doris/thirdparty/installed/lib/libsnappy.a  /Programs/doris/thirdparty/installed/lib/libcurl.a  /Programs/doris/thirdparty/installed/lib/liblz4.a  /Programs/doris/thirdparty/installed/lib/libthrift.a  /Programs/doris/thirdparty/installed/lib/libthriftnb.a  /Programs/doris/thirdparty/installed/lib/libevent.a  /Programs/doris/thirdparty/installed/lib/libevent_pthreads.a  /Programs/doris/thirdparty/installed/lib/libbz2.a  /Programs/doris/thirdparty/installed/lib/libz.a  /Programs/doris/thirdparty/installed/lib/libcrypto.a  /Programs/doris/thirdparty/installed/lib/libssl.a  /Programs/doris/thirdparty/installed/lib/libleveldb.a  /Programs/doris/thirdparty/installed/lib/libjemalloc_doris.a  /Programs/doris/thirdparty/installed/lib/libjemalloc.a  /Programs/doris/thirdparty/installed/lib64/libgrpc++_reflection.a  /Programs/doris/thirdparty/installed/lib64/libgrpc.a  /Programs/doris/thirdparty/installed/lib64/libgrpc++.a  /Programs/doris/thirdparty/installed/lib64/libgrpc++_unsecure.a  /Programs/doris/thirdparty/installed/lib64/libgpr.a  /Programs/doris/thirdparty/installed/lib64/libupb.a  /Programs/doris/thirdparty/installed/lib64/libcares.a  /Programs/doris/thirdparty/installed/lib64/libaddress_sorting.a  /Programs/doris/thirdparty/installed/lib64/libz.a  /Programs/doris/thirdparty/installed/lib64/libbrotlicommon.a  /Programs/doris/thirdparty/installed/lib64/libbrotlidec.a  /Programs/doris/thirdparty/installed/lib64/libbrotlienc.a  /Programs/doris/thirdparty/installed/lib64/libzstd.a  /Programs/doris/thirdparty/installed/lib64/libarrow.a  /Programs/doris/thirdparty/installed/lib64/libarrow_flight.a  /Programs/doris/thirdparty/installed/lib64/libarrow_flight_sql.a  /Programs/doris/thirdparty/installed/lib64/libparquet.a  /Programs/doris/thirdparty/installed/lib64/libbrpc.a  /Programs/doris/thirdparty/installed/lib/librocksdb.a  /Programs/doris/thirdparty/installed/lib/libsasl2.a  /Programs/doris/thirdparty/installed/lib/librdkafka++.a  /Programs/doris/thirdparty/installed/lib/librdkafka.a  /Programs/doris/thirdparty/installed/lib/libs2.a  /Programs/doris/thirdparty/installed/lib/libbitshuffle.a  /Programs/doris/thirdparty/installed/lib/libroaring.a  /Programs/doris/thirdparty/installed/lib/libfmt.a  /Programs/doris/thirdparty/installed/lib/libcctz.a  /Programs/doris/thirdparty/installed/lib64/libaws-cpp-sdk-core.a  /Programs/doris/thirdparty/installed/lib64/libaws-cpp-sdk-s3.a  /Programs/doris/thirdparty/installed/lib64/libaws-cpp-sdk-transfer.a  /Programs/doris/thirdparty/installed/lib64/libaws-cpp-sdk-s3-crt.a  /Programs/doris/thirdparty/installed/lib64/libaws-crt-cpp.a  /Programs/doris/thirdparty/installed/lib64/libaws-c-cal.a  /Programs/doris/thirdparty/installed/lib64/libaws-c-auth.a  /Programs/doris/thirdparty/installed/lib64/libaws-c-compression.a  /Programs/doris/thirdparty/installed/lib64/libaws-c-common.a  /Programs/doris/thirdparty/installed/lib64/libaws-c-event-stream.a  /Programs/doris/thirdparty/installed/lib64/libaws-c-io.a  /Programs/doris/thirdparty/installed/lib64/libaws-c-http.a  /Programs/doris/thirdparty/installed/lib64/libaws-c-mqtt.a  /Programs/doris/thirdparty/installed/lib64/libaws-checksums.a  /Programs/doris/thirdparty/installed/lib64/libaws-c-s3.a  /Programs/doris/thirdparty/installed/lib64/libminizip.a  /Programs/doris/thirdparty/installed/lib64/libsimdjson.a  /Programs/doris/thirdparty/installed/lib64/libidn.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_common.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_exporter_zipkin_trace.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_resources.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_version.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_exporter_ostream_span.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_trace.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_http_client_curl.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_exporter_otlp_http.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_exporter_otlp_http_client.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_otlp_recordable.a  /Programs/doris/thirdparty/installed/lib64/libopentelemetry_proto.a  /Programs/doris/thirdparty/installed/lib64/libxml2.a  /Programs/doris/thirdparty/installed/lib64/liblzma.a  /Programs/doris/thirdparty/installed/lib/libgsasl.a  /Programs/doris/thirdparty/installed/lib/libkrb5support.a  /Programs/doris/thirdparty/installed/lib/libkrb5.a  /Programs/doris/thirdparty/installed/lib/libcom_err.a  /Programs/doris/thirdparty/installed/lib/libk5crypto.a  /Programs/doris/thirdparty/installed/lib/libgssapi_krb5.a  /Programs/doris/thirdparty/installed/lib/libbfd.a  /Programs/doris/thirdparty/installed/lib/libiberty.a  /Programs/doris/thirdparty/installed/lib/libintl.a  /Programs/doris/thirdparty/installed/lib/libhdfs3.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_commandlineflag.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_commandlineflag_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_config.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_marshalling.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_parse.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_private_handle_accessor.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_program_name.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_reflection.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_usage.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_usage_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_distributions.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_distribution_test_util.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_platform.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_pool_urbg.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen_hwaes.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen_hwaes_impl.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen_slow.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_seed_material.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_seed_gen_exception.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_seed_sequences.a  /Programs/doris/thirdparty/installed/lib/libabsl_status.a  /Programs/doris/thirdparty/installed/lib/libabsl_statusor.a  /Programs/doris/thirdparty/installed/lib/libabsl_strerror.a  /Programs/doris/thirdparty/installed/lib64/libboost_container.a  /Programs/doris/thirdparty/installed/lib/libbfd.a  /Programs/doris/thirdparty/installed/lib/libiberty.a  /Programs/doris/thirdparty/installed/lib/libintl.a  /Programs/doris/thirdparty/installed/lib/libkrb5support.a  /Programs/doris/thirdparty/installed/lib/libkrb5.a  /Programs/doris/thirdparty/installed/lib/libcom_err.a  /Programs/doris/thirdparty/installed/lib/libgssapi_krb5.a  /Programs/doris/thirdparty/installed/lib/libk5crypto.a  bin/liborc.a  bin/libic.a  bin/libclucene-core-static.a  bin/libclucene-shared-static.a  bin/libclucene-contribs-lib.a  -lapple_nghttp2  -lresolv  -liconv  -framework CoreFoundation  -framework CoreGraphics  -framework CoreText  -framework Foundation  -framework SystemConfiguration  -framework Security  -Wl,-U,_MallocExtension_ReleaseFreeMemory  /Programs/doris/thirdparty/installed/lib/libabsl_flags.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_marshalling.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_reflection.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_config.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_program_name.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_private_handle_accessor.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_commandlineflag.a  /Programs/doris/thirdparty/installed/lib/libabsl_flags_commandlineflag_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_hash.a  /Programs/doris/thirdparty/installed/lib/libabsl_city.a  /Programs/doris/thirdparty/installed/lib/libabsl_low_level_hash.a  /Programs/doris/thirdparty/installed/lib/libabsl_raw_hash_set.a  /Programs/doris/thirdparty/installed/lib/libabsl_hashtablez_sampler.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_pool_urbg.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen_hwaes.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen_hwaes_impl.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen_slow.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_platform.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_seed_gen_exception.a  /Programs/doris/thirdparty/installed/lib/libabsl_random_internal_seed_material.a  /Programs/doris/thirdparty/installed/lib/libabsl_status.a  /Programs/doris/thirdparty/installed/lib/libabsl_str_format_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_strerror.a  /Programs/doris/thirdparty/installed/lib/libabsl_cord.a  /Programs/doris/thirdparty/installed/lib/libabsl_bad_optional_access.a  /Programs/doris/thirdparty/installed/lib/libabsl_cordz_info.a  /Programs/doris/thirdparty/installed/lib/libabsl_cord_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_cordz_functions.a  /Programs/doris/thirdparty/installed/lib/libabsl_exponential_biased.a  /Programs/doris/thirdparty/installed/lib/libabsl_cordz_handle.a  /Programs/doris/thirdparty/installed/lib/libabsl_synchronization.a  /Programs/doris/thirdparty/installed/lib/libabsl_graphcycles_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_time.a  /Programs/doris/thirdparty/installed/lib/libabsl_civil_time.a  /Programs/doris/thirdparty/installed/lib/libabsl_time_zone.a  -Xlinker -framework -Xlinker CoreFoundation  /Programs/doris/thirdparty/installed/lib/libabsl_crc_cord_state.a  /Programs/doris/thirdparty/installed/lib/libabsl_crc32c.a  /Programs/doris/thirdparty/installed/lib/libabsl_crc_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_crc_cpu_detect.a  /Programs/doris/thirdparty/installed/lib/libabsl_stacktrace.a  /Programs/doris/thirdparty/installed/lib/libabsl_symbolize.a  /Programs/doris/thirdparty/installed/lib/libabsl_malloc_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_debugging_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_demangle_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_strings.a  /Programs/doris/thirdparty/installed/lib/libabsl_int128.a  /Programs/doris/thirdparty/installed/lib/libabsl_strings_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_base.a  /Programs/doris/thirdparty/installed/lib/libabsl_spinlock_wait.a  /Programs/doris/thirdparty/installed/lib/libabsl_throw_delegate.a  /Programs/doris/thirdparty/installed/lib/libabsl_bad_variant_access.a  /Programs/doris/thirdparty/installed/lib/libabsl_raw_logging_internal.a  /Programs/doris/thirdparty/installed/lib/libabsl_log_severity.a  /Programs/doris/thirdparty/installed/lib/libprotobuf.a  /Programs/doris/thirdparty/installed/lib/libz.a  /Programs/doris/thirdparty/installed/lib/libsnappy.a  /Programs/doris/thirdparty/installed/lib/liblz4.a  /Programs/doris/thirdparty/installed/lib/libzstd.a  bin/libclucene-core-static.a  bin/libclucene-shared-static.a && :
ld: warning: ignoring duplicate libraries: '/Programs/doris/thirdparty/installed/lib/libabsl_flags.a', '/Programs/doris/thirdparty/installed/lib/libabsl_flags_commandlineflag.a', '/Programs/doris/thirdparty/installed/lib/libabsl_flags_commandlineflag_internal.a', '/Programs/doris/thirdparty/installed/lib/libabsl_flags_config.a', '/Programs/doris/thirdparty/installed/lib/libabsl_flags_internal.a', '/Programs/doris/thirdparty/installed/lib/libabsl_flags_marshalling.a', '/Programs/doris/thirdparty/installed/lib/libabsl_flags_private_handle_accessor.a', '/Programs/doris/thirdparty/installed/lib/libabsl_flags_program_name.a', '/Programs/doris/thirdparty/installed/lib/libabsl_flags_reflection.a', '/Programs/doris/thirdparty/installed/lib/libabsl_random_internal_platform.a', '/Programs/doris/thirdparty/installed/lib/libabsl_random_internal_pool_urbg.a', '/Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen.a', '/Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen_hwaes.a', '/Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen_hwaes_impl.a', '/Programs/doris/thirdparty/installed/lib/libabsl_random_internal_randen_slow.a', '/Programs/doris/thirdparty/installed/lib/libabsl_random_internal_seed_material.a', '/Programs/doris/thirdparty/installed/lib/libabsl_random_seed_gen_exception.a', '/Programs/doris/thirdparty/installed/lib/libabsl_status.a', '/Programs/doris/thirdparty/installed/lib/libabsl_strerror.a', '/Programs/doris/thirdparty/installed/lib/libbfd.a', '/Programs/doris/thirdparty/installed/lib/libcom_err.a', '/Programs/doris/thirdparty/installed/lib/libgssapi_krb5.a', '/Programs/doris/thirdparty/installed/lib/libiberty.a', '/Programs/doris/thirdparty/installed/lib/libintl.a', '/Programs/doris/thirdparty/installed/lib/libk5crypto.a', '/Programs/doris/thirdparty/installed/lib/libkrb5.a', '/Programs/doris/thirdparty/installed/lib/libkrb5support.a', '/Programs/doris/thirdparty/installed/lib/liblz4.a', '/Programs/doris/thirdparty/installed/lib/libprotobuf.a', '/Programs/doris/thirdparty/installed/lib/libsnappy.a', '/Programs/doris/thirdparty/installed/lib/libz.a', 'bin/libclucene-core-static.a', 'bin/libclucene-shared-static.a'
ld: archive member '/' not a mach-o file in '/Programs/doris/thirdparty/installed/lib64/libabsl_flags.a'
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

### What You Expected?

Build BE successfully.

### How to Reproduce?

```shell
BUILD_TYPE=ASAN ./build.sh --be -j 8 --clean
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/be/CMakeLists.txt
+++ b/be/CMakeLists.txt
@@ -506,34 +506,14 @@ endif()
 if (absl_FOUND)
     set(COMMON_THIRDPARTY
         ${COMMON_THIRDPARTY}
-        absl::flat_hash_set
-        absl::str_format
-        absl::flags
-        absl::flags_commandlineflag
-        absl::flags_commandlineflag_internal
-        absl::flags_config
         absl::flags_internal
-        absl::flags_marshalling
-        absl::flags_parse
-        absl::flags_private_handle_accessor
-        absl::flags_program_name
         absl::flags_reflection
-        absl::flags_usage
-        absl::flags_usage_internal
-        absl::random_distributions
-        absl::random_internal_distribution_test_util
-        absl::random_internal_platform
         absl::random_internal_pool_urbg
         absl::random_internal_randen
-        absl::random_internal_randen_hwaes
-        absl::random_internal_randen_hwaes_impl
-        absl::random_internal_randen_slow
-        absl::random_internal_seed_material
-        absl::random_seed_gen_exception
-        absl::random_seed_sequences
+        absl::spinlock_wait
         absl::status
         absl::statusor
-        absl::strerror
+        absl::strings
     )
 endif()
 

